components/base/wheeled/wheeled_base.go

```diff
@@ -125,9 +142,15 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 }
 
 // MoveStraight commands a base to drive forward or backwards  at a linear speed and for a specific distance.
-func (base *wheeledBase) MoveStraight(ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{}) error {
+// TODO RSDK-2362 check choppiness of movement when run as a remote.
+func (base *wheeledBase) MoveStraight(
+	ctx context.Context, distanceMm int, mmPerSec float64, extra map[string]interface{},
+) error {
 	ctx, done := base.opMgr.New(ctx)
 	defer done()
+
+	base.stopSensors()
```

Comment:
this is how I can stop the for loop when the other API calls interrupt spin, otherwise, there is a routine running that calls `base.Stop` once the desired heading angle is reached.